internal/files/repository.go

```diff
@@ -24,6 +46,14 @@ func FindRepositoryRootDirectory() (string, error) {
 	for dir != "." {
 		path := filepath.Join(dir, ".git")
 		fileInfo, err := os.Stat(path)
+		if err == nil && !fileInfo.IsDir() {
+			errWorktree := isGitWorktree(path)
+			if errWorktree != nil {
+				return "", errWorktree
+			}
+
+			return dir, nil
+		}
```

Comment:
Nit. Maybe we can move all this logic to a `isGitRootDir(dir string) (bool, err)`  function, that checks both known conditions.